src/ReportGenerator.Core/Parser/DynamicCodeCoverageParser.cs

```diff
@@ -187,7 +187,9 @@ private static CodeFile ProcessFile(XElement module, string fileId, ClassWithNam
             var linesOfFile = methods
                 .Elements("ranges")
                 .Elements("range")
+                .Where(r => r.Attribute("source_id").Value == fileId)
```

Comment:
This prevent the tool from applying the same coverage for different files invoked in the method being processed. It should only apply the associated coverage for the file.